hms-dbmi-cellenics/ui
# [BIOMAGE-1764] - Fix Error when going back from single-sample DE

# Description
#### Issue
Clicking "Go Back" from the DE table causes an error to happen. This happens in the piece of code to automatically select the only sample available for the dataset.  The piece of code does dispatch action to update the selected sample, but uses direct assignment, which causes the error.

#### Solution
Dispatch action to update the selected sample correctly.

#### To test
To reproduce the bug :

- Launch a project with a single sample
- Go to Data Management, launch DE with any options
- Click "Go Back"
- This should throw an error, causing an "it's not you, it's us" error page to show up.

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1764

#### Link to staging deployment URL (or set N/A)
https://ui-agi-ui655.scp-staging.biomage.net/

#### Links to any PRs or resources related to this PR
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR

### Manual/unit testing

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.


### Documentation updates

### Optional
